### PR TITLE
[Core] Linter rule for service_name.json

### PR DESCRIPTION
**Pull request by qwordy**
**Description**

Check format of service_name.json. Command and AzureServiceName are required. Others are optional.
Each highest level command group should have reference in service_name.json.

https://ceapex.visualstudio.com/Engineering/_workitems/edit/327925
https://docs.microsoft.com/en-us/cli/azure/azure-cli-reference-for-network?view=azure-cli-latest

[Core] Linter rule for service_name.json

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yungezz** [comment]
hi @qwordy, as offline synced, pls share with the team context of the rule, what's it for before merging it. thanks

**Comment by qwordy** [comment]
> 
> 
> hi @qwordy, as offline synced, pls share with the team context of the rule, what's it for before merging it. thanks

OK

**Comment by qwordy** [comment]
Test Extensions Loading Python36 keeps failing.

**Review by kairu-ms — APPROVED** [review]
It's clear now.

File: src/azure-cli/service_name.json [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+  },
+  {
+    "Command": "az cdn",
+    "AzureServiceName": "Content Delivery Network",
+    "URL": "https://docs.microsoft.com/azure/cdn/"
+  },
+  {
+    "Command": "az cloud",
+    "AzureServiceName": "Azure CLI",
+    "URL": "https://docs.microsoft.com/azure/cloud-services/"
```
Comment (fengzhou-msft) [inline src/azure-cli/service_name.json:115]: `az cloud` is not for Azure Cloud Services.

File: scripts/ci/service_name.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+def main():
+    az_cli = AzCli(cli_name='az',
+                   commands_loader_cls=MainCommandsLoader,
+                   invocation_cls=AzCliCommandInvoker,
+                   parser_cls=AzCliCommandParser,
+                   help_cls=AzCliHelp)
+    with patch('getpass.getuser', return_value='your_system_user_login_name'):
+        create_invoker_and_load_cmds_and_args(az_cli)
+    help_files = get_all_help(az_cli)
+    high_command_set = set()
```
Comment (jiasli) [inline scripts/ci/service_name.py:30]: What is `high`?

File: src/azure-cli/service_name.json [hunk 3]
```diff
@@ -0,0 +1,517 @@
+[
+  {
+    "Command": "az account",
+    "AzureServiceName": "Azure CLI",
+    "URL": "https://docs.microsoft.com/azure/cost-management-billing/manage/create-subscription"
```
Comment (jiasli) [inline src/azure-cli/service_name.json:5]: `az account` is not for creating subscription.

File: scripts/ci/service_name.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+Each highest level command group should have reference in service_name.json.
+"""
+import json
+
+from azure.cli.core import MainCommandsLoader, AzCli
+from azure.cli.core._help import AzCliHelp
+from azure.cli.core.commands import AzCliCommandInvoker
+from azure.cli.core.file_util import create_invoker_and_load_cmds_and_args, get_all_help
+from azure.cli.core.parser import AzCliCommandParser
+from mock import patch
```
Comment (jiasli) [inline scripts/ci/service_name.py:18]: Per https://pypi.org/project/mock/,

> mock is now part of the Python standard library, available as `unittest.mock` in Python 3.3 onwards.

File: scripts/ci/service_name.py (hunk 2 above)
Comment (qwordy) [reply scripts/ci/service_name.py:30]: `high` means `left`. The first word after `az`.

File: scripts/ci/service_name.py (hunk 4 above)
Comment (qwordy) [reply scripts/ci/service_name.py:18]: You are right

**Review by jiasli — APPROVED** [review]

**Pull request merged**